mpas_analysis/sea_ice/climatology_map_risk_index_outcome.py

```diff
@@ -0,0 +1,359 @@
+
+    def __init__(self, config, mpas_climatology_task, hemisphere,
+                 control_config=None):
+        """
+        Construct the analysis task.
+
+        Parameters
+        ----------
+        config : mpas_tools.config.MpasConfigParser
+            Configuration options
+
+        mpas_climatology_task : mpas_analysis.shared.climatology.MpasClimatologyTask
+            The task that produced the climatology to be remapped and plotted
+
+        hemisphere : {'NH', 'SH'}
+            The hemisphere to plot
+
+        control_config : mpas_tools.config.MpasConfigParser, optional
+            Configuration options for a control run (if any)
+        """
+        # Authors
+        # -------
+        # Gennaro D'Angelo, Milena Veneziani
+
+        task_name = f'climatologyMapRiskIndexOutcome{hemisphere}'
+
+        field_name = 'RiskIndexOutcome'
+
+        tags = ['climatology', 'horizontalMap', field_name]
+        if hemisphere == 'NH':
+            tags = tags + ['arctic']
+        else:
+            tags = tags + ['antarctic']
+
+        # call the constructor from the base class (AnalysisTask)
+        super().__init__(config=config, taskName=task_name,
+            componentName='seaIce', tags=tags)
+
+        self.mpas_climatology_task = mpas_climatology_task
+
+        section_name = self.taskName
+
+        if hemisphere == 'NH':
+            hemisphere_long= 'Northern'
+        else:
+            hemisphere_long= 'Southern'
+
+        # read in what seasons we want to plot
+        seasons = config.getexpression(section_name, 'seasons')
+
+        if len(seasons) == 0:
+            raise ValueError(f'config section {section_name} does not contain '
+                             f'a valid list of seasons')
+
+        comparison_grid_names = config.getexpression(section_name,
+                                                   'comparisonGrids')
+
+        if len(comparison_grid_names) == 0:
+            raise ValueError(f'config section {section_name} does not contain '
+                             f'a valid list of comparison grids')
+
+        # read in what polar class of vessel we want to plot
+        polarclass = config.getexpression(section_name, 'polarClass')
+
+        if polarclass < 0 or polarclass > 12:
+            raise ValueError(f'config section {section_name} does not contain '
+                             f'a valid instance of Polar Class')
+
+        # convert to 0-based array index
+        polarclass = np.int_(polarclass) - 1
+
+        # read in table of Risk Index Values
+        #riv_csv = 'riv_MSC.1_Circ.1519_6_June_2016.csv'
+        riv_csv = build_obs_path(config, 'seaIce',
+                                 relativePathOption='riv{}'.format(hemisphere),
+                                 relativePathSection=section_name)
+        IceClassLabels = np.genfromtxt(riv_csv, delimiter=',', skip_header=1,
+                                       dtype=str, usecols=(0,))
```

Comment:
The problem here is that if `riv_csv` is missing, MPAS-Analysis will crash completely.  That's not a good state to be in.  This file needs to be read and used in a `run_task()` method, not in a constructor.  As this is currently written, even if someone tries to disable this analysis, they are required to have this CSV file in order to run other analysis.